src/goMain.ts

```diff
@@ -79,12 +87,13 @@ export function activate(ctx: vscode.ExtensionContext): void {
 								}
 							});
 						}
-						ctx.globalState.update('goVersion', currVersionString);
+						toolsGoInfo[toolsGopath].version = currVersionString;
```

Comment:
The global state gets updated before waiting for this promise to be resolved, so the current version will never get updated in the global state

Reply:
One way around is to check for goroot as well as the version only after the promise returned by `getGoVersion` is resolved